Related to #1112:Add filter for images, videos

vibhcool opened the pull request:
This PR adds feature to :- 
1) searches for tweets with images if we set `filter=image` 
2) for tweets with videos if we set `filter=image`
3) and for both if we set `filter=image,video`

This PR:-

1) Partially fixes issue #1112 , doesn't work for cache(will push when this gets accepted, working on this)
2) outputs the total number of tweets with constraint(image or video) from total count of tweets. As loklak doesn't output more than 100 tweets due to performance issues.

Examples :- 
http://localhost:9000/api/search.json?q=video&filter=video,image&source=twitter
http://localhost:9000/api/search.json?q=paper&filter=image&source=twitter
http://localhost:9000/api/search.json?q=car&filter=video&source=twitter

djmgit commented on src/org/loklak/harvester/TwitterScraper.java:
```diff
@@ -290,7 +348,30 @@ public static String prepareSearchURL(final String query) {
+                place_name = "";
+                continue;
+            }
+
```
@vibhcool please indent this if-block

kavithaenair commented on src/org/loklak/harvester/TwitterScraper.java:
```diff
@@ -165,24 +208,39 @@ public static String prepareSearchURL(final String query) {
         Set<String> images = new LinkedHashSet<>();
         Set<String> videos = new LinkedHashSet<>();
-        String place_id = "", place_name = "";
+        String place_id;
```
This can be initialized here instead of doing it in line no 235 and 236. Line 235 is also not indented properly.

kavithaenair replied:
The image filter is showing emojis (images) if they are in the username.
For this:
http://localhost:9000/api/search.json?q=yolo&filter=image&source=twitter
the status link is:
https://twitter.com/xeooop/status/864815541197742081
the image link shown here is:
https://abs.twimg.com/emoji/v2/72x72/2744.png
But here the image shown is a character or emoji used in the username.
Is `filter=image` working fine?

vibhcool replied:
this point about wrong images being shown is different issue related to #1135 i.e. storing wrong images.
filter=image just takes tweets which has images.

singhpratyush commented:
@vibhcool: Please write about how this patch works in the description.

vibhcool commented:
@singhpratyush you are somewhat inferring opposite, but I got the point, that is about using better approach. will fix it . thanks 👍

vibhcool commented:
@djmgit , will add the test for this

singhpratyush commented:
@vibhcool: Could you please give the query for which it is showing non-image or non-video results?

Also, the issue you mentioned says that the search should be of type `github \image` and not `q=github&filter=image`. So you may directly check if those terms exist in the search query instead of introducing a new parameter.

vibhcool commented:
updated PR

1) Updated test for testing prepareSearchUrl method.
2) Some indentation and minor fixes in TwitterScraperTest.
3) Added Better approach for filtering tweets with videos.
 - I am directly fetching videos by setting `f=videos` in twitter url like `https://twitter.com/search?f=videos&vertical=default&q=paris&src=typd`
 - Not using this method for fetching images as on setting `f=images` , only images are fetched instead of fetching complete tweets having images.
4) Working on test for searching for tweets with filters.

vibhcool commented:
@daminisatya working on it
